### ISSUE TITLE
Rename `assets` to `fonts`

### ISSUE THREAD

**Issue by grabbou** [rnpm/rnpm#18]
This is somehow because I've misunderstood the use-case of that when we were discussing it @Kureev. With the new static system, the only real use-case is to copy fonts (which is valid because they cannot be referenced as images using file system paths):

In the current `copyAssets` action:

iOS:
1. Create group in XCode project called `Fonts` if it's not yet there (name can be changed with custom config just in case) and attach these fonts to that group (references node_modules/project/fonts/a.ttf etc)
3. Modify info.plist

Android:
1. Just copy stuff from `fonts`  to `src/main/assets/fonts`

**Comment by grabbou** [comment]
And obviously rename copyAssets to copyFonts.

**Comment by Kureev** [comment]
I think we can be a bit more flexible about it: I propose to replace `assets` by custom copy script. It'll looks like this:
```js
copy: ['somefont.ttf', 'somesound.ogg']
```

**Comment by grabbou** [comment]
But for the iOS, it's not really a copy. And you handle fonts differently than `ogg`. For iOS fonts to work, you just have to create a group folder that references node_modules/project/fonts location and attach these fonts to Info.plist. That's all.

So since these actions will be highly type specific, I'd say let's leave the `assets` thing, but once we get an array of all files, just group them by type (e.g. fonts, images, sounds) and decide what to do based on that.

**Comment by Kureev** [comment]
Agree :+1:

**Comment by grabbou** [comment]
Cool, I'll send over PR later today :)

**Comment by Kureev** [comment]
Sounds like an interesting task btw. Looking forward to see your PR

**Comment by grabbou** [comment]
I am just not sure about that one
> but once we get an array of all files, just group them by type (e.g. fonts, images, sounds) and decide what to do based on that.

This is highly error - prone to detect all extensions... I'd better change the config of rnpm from assets to either
```
assets: {fonts: [], images: []}
```
or
```
fonts: [],
images: []
```

**Comment by cbrevik** [comment]
I am wondering if I misunderstood the the assets folder linking at first (as per our discussion in at reactiflux).

Please correct me if I'm wrong; but the use case you guys are trying to solve here is if for example Kureev wanted to add custom fonts to his react-native-side-menu project, then he would add the following to the package.json in that project: ```"rnpm": {
    "assets": ["fonts"]
  }```

Then when I do ```npm install react-native-side-menu --save``` in my React Native app, and then ```rnpm link```, then whatever assets he has in the folder will be linked up in my Xcode-project.

What I thought at first though, was that I could **also** define custom fonts in my own app (i.e. not anything I'm linking in from a third-party project), and add that rnpm clause to the package.json of my project, and it would be added to the Xcode- and Android-projects of my app. 

But that is not a use case you are trying to solve right?

**Comment by Kureev** [comment]
I'm currently developing plugin system for that. You'll be able to create a plugin for `rnpm` which will solve your issue :wink:

**Comment by grabbou** [comment]
That's very interesting idea that can be probably incorporated into post link hook once it's done. Would you mind opening separate issue for that so we keep track of it? It's actually super simple to do and will be made up from 3 or 4 lines, so we might add it to link command anyway as it's going to end up in a separate package anyway.

**Comment by grabbou** [comment]
This is now being implemented in https://github.com/rnpm/rnpm-plugin-link/pull/1

**Comment by grabbou** [comment]
Implemented in 1.1.0.
